src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -318,7 +318,27 @@ public async Task InitiateBrokerHandshakeAsync(Activity callerActivity)
                             return;
                         }
 
-                        throw new MsalClientException("Could not negotiate protocol version with broker.");
+                        dynamic errorResult = JObject.Parse(bundleResult?.GetString(BrokerConstants.BrokerResultV2));
+                        string errorCode = null;
+                        string errorDescription = null;
+
+                        if (!string.IsNullOrEmpty(errorResult))
+                        {
+                            errorCode = errorResult[BrokerResponseConst.BrokerErrorCode]?.ToString();
+                            errorDescription = errorResult[BrokerResponseConst.BrokerErrorMessage]?.ToString();
```

Comment:
>errorDescription [](start = 28, length = 16)

nit: wouldn't this be the error Message, and the below thing you log be the error description?